Core/Core/Planner/CalendarToDo/ViewModel/EditCalendarToDoViewModel.swift

```diff
@@ -49,46 +85,79 @@ final class EditCalendarToDoViewModel: ObservableObject {
     // MARK: - Input
 
     let didTapCancel = PassthroughSubject<Void, Never>()
-    let didTapAdd = PassthroughSubject<Void, Never>()
+    let didTapSave = PassthroughSubject<Void, Never>()
 
     // MARK: - Private
 
+    private let mode: Mode
     private let toDoInteractor: CalendarToDoInteractor
     private let calendarListProviderInteractor: CalendarFilterInteractor
-    private var subscriptions = Set<AnyCancellable>()
     private var selectedCalendar = CurrentValueSubject<CDCalendarFilterEntry?, Never>(nil)
+    private var isTouched: Bool = false
+
+    private var subscriptions = Set<AnyCancellable>()
 
     // MARK: - Init
 
     init(
+        plannable: Plannable? = nil,
         toDoInteractor: CalendarToDoInteractor,
         calendarListProviderInteractor: CalendarFilterInteractor,
         completion: @escaping (PlannerAssembly.Completion) -> Void
     ) {
         self.toDoInteractor = toDoInteractor
         self.calendarListProviderInteractor = calendarListProviderInteractor
 
-        // end of today, to match default web behaviour
-        date = .now.endOfDay()
+        if let plannable {
+            mode = .edit(id: plannable.id)
+        } else {
+            mode = .add
+        }
+
+        title = plannable?.title ?? ""
+        date = plannable?.date ?? Clock.now.endOfDay() // end of today, to match default web behaviour
+        details = plannable?.details ?? ""
+
+        subscribeIsTouched(to: $title)
```

Comment:
I'm not sure if there's a better name for this logic instead of `isTouched` but I didn't get its meaning for the first time

Reply:
updated to `isFieldsTouched`, as discussed